tokenserver/scripts/purge_old_records.py

```diff
@@ -58,11 +58,17 @@ def purge_old_records(config_file, grace_period=-1, max_per_loop=10,
                 }
                 rows = list(backend.get_old_user_records(service, **kwds))
                 for row in rows:
-                    logger.info("Purging uid %s on %s", row.uid, row.node)
-                    if row.node is not None:
-                        delete_service_data(config, service, row,
-                                            timeout=request_timeout)
-                    backend.delete_user_record(service, row.uid)
+                    # Don't attempt to purge data from downed nodes.
+                    # Instead wait for them to either come back up or to be
+                    # compeltely removed from service.
+                    if row.node is None:
+                        backend.delete_user_record(service, row.uid)
+                    elif not row.downed:
+                        logger.info("Purging uid %s on %s", row.uid, row.node)
+                        if row.node is not None:
```

Comment:
nit: the elif confirms row.node is not None, this check isn't necessary

Reply:
thanks, nice catch, this is left over from a previous factoring